fem/bilininteg.hpp

```diff
@@ -3168,6 +3182,7 @@ class DGTraceIntegrator : public BilinearFormIntegrator
 
 private:
    Vector shape1, shape2;
+   Vector tr_shape1, te_shape1, tr_shape2, te_shape2;
```

Comment:
I see that for clarity you want to use these specific names for the shape function vectors but you could reuse `shape1` and `shape2` and just add only two extra

Reply:
I know, but it is for clarity and depending on the usage, either the first pair or the second will be empty vectors, so a minimal overhead. If you are really concerned about the memory consumption, we may do it like this `Vector &shape1{tr_shape1}, &shape2{tr_shape2};`, but I would definitely prefer to keep different names.